### DISCUSSION TITLE
Ledger and topics

### DISCUSSION THREAD

**Discussion by ub-3** [apache/pulsar#23034] [Q&A] [answered]
1. Can one ledger of bookkeeper relate to multiple topics? 
2. And how to find the topics to which it belongs by ledger id.

**Comment by sbourkeostk** [comment]
1. No.
2. It's in the ledger metadata which is stored in the zookeeper. In the example below the customMetadata entry `pulsar/managed-ledger` points to the managed ledger. `offers/served/persistent/daily-partition-2` represents the pulsar topic: `persistent://offers/served/daily-partition-2`

```
quorumSize: 2
ensembleSize: 2
length: 2915
lastEntryId: 9
state: CLOSED
segment {
  ensembleMember: "10.0.1.35:3181"
  ensembleMember: "10.0.1.31:3181"
  firstEntryId: 0
}
digestType: CRC32C
password: ""
ackQuorumSize: 2
customMetadata {
  key: "component"
  value: "managed-ledger"
}
customMetadata {
  key: "pulsar/managed-ledger"
  value: "offers/served/persistent/daily-partition-2"
}
customMetadata {
  key: "application"
  value: "pulsar"
}
```

**Reply by ub-3** [reply]
The problem is that I do not know which topic a ledger belongs to and I want to find this topic.

**Comment by sbourkeostk** [comment, marked as answer]
The above data is from the ledger metadata. You can get this metadata based off the ledger id:
[bookkeeper.getLedgerMetadata(ledger_id)](https://bookkeeper.apache.org/docs/latest/api/javadoc/org/apache/bookkeeper/client/BookKeeper.html#getLedgerMetadata(long))
